### PR TITLE
Fixes Kudzu growing in space

### LINKED ISSUE #36479: Kudzu isn't supposed to grow into space?
image

### PR DESCRIPTION
**Pull request by kachnov**
fixes #36479

### CODE REVIEW

**Comment by chesse20** [comment]
its a feature not a bug. Down Vote

**Comment by Qustinnus** [comment]
how is it a feature if there is an issue that tags it as a bug, who are you to say it's a feature.

**Comment by ExcessiveUseOfCobblestone** [comment]
why would you check if the turf is space AFTER you attempt to enter it?

**Comment by kachnov** [comment]
@ExcessiveUseOfCobblestone stepturf.Enter(src) is just a check to see if the vine can go on that turf. Its probably more efficient this way because we have the chance to short circuit and not do the isspaceturf() typecheck

**Comment by ExcessiveUseOfCobblestone** [comment]
or you could do it before the mutations so it doesn't run through that loop

**Comment by kachnov** [comment]
code updated

File: code/modules/events/spacevine.dm [hunk 1]
```diff
@@ -510,11 +510,11 @@
 /obj/structure/spacevine/proc/spread()
 	var/direction = pick(GLOB.cardinals)
 	var/turf/stepturf = get_step(src,direction)
-	for(var/datum/spacevine_mutation/SM in mutations)
-		SM.on_spread(src, stepturf)
-		stepturf = get_step(src,direction) //in case turf changes, to make sure no runtimes happen
-	if(!locate(/obj/structure/spacevine, stepturf))
-		if(stepturf.Enter(src))
+	if (stepturf.Enter(src) && !isspaceturf(stepturf))
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/events/spacevine.dm]: OCD triggering in 5...

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by AlexTheSysop42** [comment]
have the PR body say "fixes #36479" so it autocloses

**Comment by Razharas** [comment]
Instead of invisible wall it should kill the vine upon trying to enter the space turf
So add the kill proc to the entered of space vine piece on space entrance

**Comment by kachnov** [comment]
what kill proc

**Review by AnturK — DISMISSED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
@AnturK

**Pull request merged**